pydatastructs/linear_data_structures/algorithms.py

```diff
@@ -787,3 +788,65 @@ def longest_common_subsequence(seq1: OneDimensionalArray, seq2: OneDimensionalAr
                     check_mat[i][j] = check_mat[i][j-1]
 
     return OneDimensionalArray(seq1._dtype, check_mat[row][col][-1])
+
+def longest_increasing_subsequence(seq: OneDimensionalArray) -> int:
```

Comment:
The function currently returns only the length. There should be a possibility of obtaining the longest increasing sub-sequence as well.

For example, if `ODA([1, 2, -1, 3, 4, 5])` is the input then, `(ODA[1, 2, 3, 4, 5], 5)` should be the output.

Reply:
Please refer to this for restoring the sequence: https://cp-algorithms.com/sequences/longest_increasing_subsequence.html#toc-tgt-8